lib/puppet/rest/routes.rb

```diff
@@ -34,11 +34,14 @@ def self.get_certificate(client, name)
     # HTTP client. Accepts a block to stream responses to disk.
     # @param [Puppet::Rest::Client] client the HTTP client to use to make the request
     # @param [String] name the crl to fetch
+    # @param [String] last_modified the httpdate of when the agent crl was last modified
     # @raise [Puppet::Rest::ResponseError] if the response status is not OK
     # @return nil
-    def self.get_crls(client, name, &block)
+    def self.get_crls(client, name, last_modified, &block)
```

Comment:
Maybe better to do the `require 'time'` here and just accept a regular Time object for `last_modified`, performing the `.httpdate` conversion inside this class when setting `If-Modified-Since` header?

Reply:
Yeah, I went back and forth on that.  I think that may be better since httpdate is probably not used much outside of headers.